### PR TITLE
feat: [PIE-4739]: Allowed Values - on error, input value should be re…

### PR DESCRIPTION
**Pull request by ishantmehtaharness**
…tained

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10591)

**Comment by ishantmehtaharness** [comment]
retrigger jest

**Comment by ishantmehtaharness** [comment]
retrigger typecheck

**Comment by ishantmehtaharness** [comment]
retrigger typecheck

File: src/modules/10-common/components/ConfigureOptions/__tests__/AllowedValuesField.test.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        />
+      </TestWrapper>
+    )
+    const inputEl = container.getElementsByTagName('input')[0]
+    fireEvent.change(inputEl, { target: { value: '20s' } })
+    // // { key: "Enter", code: 13, charCode: 13 }
+    // await fireEvent.keyDown(inputEl, { key: 'Enter', code: 13, charCode: 13 })
+    // // fireEvent.click()
+    // expect(inputEl).toMatchSnapshot()
+    //
```
Comment (ishantmehtaharness) [inline src/modules/10-common/components/ConfigureOptions/__tests__/AllowedValuesField.test.tsx:58]: This is WIP, adding more in next PR

**Review by vkbansal-harness — APPROVED** [review]

File: src/modules/10-common/components/ConfigureOptions/__tests__/AllowedValuesField.test.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      errors: {},
+      setFieldTouched: jest.fn(),
+      setFieldValue: jest.fn()
+    }
+    const { container } = render(
+      <TestWrapper>
+        <RenderField options={[]} getString={getString} isReadonly={false} formik={formik as any} />
+      </TestWrapper>
+    )
+    expect(container).toMatchSnapshot()
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/ConfigureOptions/__tests__/AllowedValuesField.test.tsx:31]: We want to write as less snapshot testing as possible. Simple assertion will do the job here, I think.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/10-common/components/ConfigureOptions/__tests__/AllowedValuesField.test.tsx (hunk 2 above)
Comment (ishantmehtaharness) [reply src/modules/10-common/components/ConfigureOptions/__tests__/AllowedValuesField.test.tsx:31]: Sure, I will have a look into it Chetan. Thanks for the feedback

**Pull request merged**